src/main/java/org/jenkinsci/plugins/orgfolder/github/MainLogic.java

```diff
@@ -119,11 +119,11 @@ public void applyOrg(OrganizationFolder of, GitHubSCMNavigator scm) throws IOExc
      */
     public void applyRepo(WorkflowMultiBranchProject item, GitHubSCMNavigator scm) throws IOException {
         if (UPDATING.get().add(item)) {
-            GitHub hub = connect(item, scm);
-            GHRepository repo = hub.getRepository(scm.getRepoOwner() + '/' + item.getName());
-
             BulkChange bc = new BulkChange(item);
             try {
+                GitHub hub = connect(item, scm);
+                GHRepository repo = hub.getRepository(scm.getRepoOwner() + '/' + item.getName());
+
```

Comment:
I suppose you are just trying to ensure that the `UPDATING` removal `finally` block cannot be skipped by a failure in `connect` or `getRepository` etc. It would be safer to actual do that directly:

```java
if (UPDATING.get().add(item)) {
    try {
        GitHub hub = connect(item, scm);
        GHRepository repo = hub.getRepository(scm.getRepoOwner() + '/' + item.getName());
        BulkChange bc = new BulkChange(item);
        try {
            // …
            bc.commit();
        } finally {
            bc.abort();
        }
    } finally {
        UPDATING.get().remove(item);
    }
}
```